skfda/representation/basis.py

```diff
@@ -326,6 +326,70 @@ def _list_to_R(self, knots):
     def _to_R(self):
         raise NotImplementedError
 
+    def _inner_matrix(self, other=None):
+
+        r"""Return the Inner Product Matrix of a pair of basis.
+
+        The Inner Product Matrix is defined as
+
+        .. math::
+            IP_{ij} = \langle\phi_i, \theta_j\rangle
+
+        where :math:`\phi_i` is the ith element of the basi and
+        :math:`\theta_j` is the jth element of the second basis.
+        This matrix helps on the calculation of the inner product
+        between objects on two basis and for the change of basis.
+
+        Args:
+            other (:class:`Basis`): Basis to compute the inner product
+            matrix. If not basis is given, it computes the matrix with
+            itself returning the Gram Matrix
+
+        Returns:
+            numpy.array: Inner Product Matrix of two basis
+
+        """
+        if other is None or self == other:
+            return self.gram_matrix()
+
+        first = self.to_basis()
+        second = other.to_basis()
+
+        inner = np.zeros((self.nbasis, other.nbasis))
+
+        for i in range(self.nbasis):
+            for j in range(other.nbasis):
+                inner[i, j] = first[i].inner_product(second[j], None, None)
+
+        return inner
+
+    def gram_matrix(self):
+
+        r"""Return the Gram Matrix of a basis
+
+        The Gram Matrix is defined as
+
+        .. math::
+            G_{ij} = \langle\phi_i, \phi_j\rangle
+
+        where :math:`\phi_i` is the ith element of the basis. This is a symmetric matrix and
+        positive-semidefinite.
+
+        Returns:
+            numpy.array: Gram Matrix of the basis.
+
+        """
+        fbasis = self.to_basis()
+
+        gram = np.zeros((self.nbasis, self.nbasis))
+
+        for i in range(fbasis.nbasis):
+            for j in range(i, fbasis.nbasis):
+                gram[i, j] = fbasis[i].inner_product(fbasis[j], None, None)
```

Comment:
This integration can be avoided for the most common cases, using an explicit formula, if it's still running slow.
For example , in the polynomial basis in `[a, b]`, 

`gram[i, j] = <t^i, t^j> = int_a^b t ^(i + j) dt = (b^(i+j+1) - a^(i+j+1)) / (i+j+1)`.

Maybe could be added as an issue override this methods when the product could be make without numeric integration.

Reply:
Add it.

Reply:
Yo pensé en abrir el issue una vez terminado como con las operaciones. Hice los casos generales y dejé para futuro los casos particulares.